### PR TITLE
Enable React Fast Refresh

### PR DESCRIPTION
**Pull request by dwightwatson**
### Type: Feature

Closes: #322

### What are the changes and their implications? :gear:

This enables Fast Refresh for Blitz apps by default.

It bumps the minimum supported version of Next.js to `9.3.7` (as-yet unreleased) as it will be the first version to properly support fast refresh (as noted in the attached issue).

This can't be merged until `9.3.7` has properly landed.

### Checklist

- [ ] Tests added for changes
- [ ] Any added terminal logging uses `packages/server/src/log.ts`

### Breaking change: No

As this is just opting-in to a new development configuration for React I wouldn't expect it to break anything.

### Other information

This is my first Blitz PR so I hope I've gone about it the right way!

### CODE REVIEW

**Comment by dwightwatson** [comment]
Failure expected as it depends on the unreleased Next.js - but I assume we can rebuild once it's been tagged.

File: package.json [hunk 1]
```diff
@@ -73,7 +73,7 @@
     "jest": "24.9.0",
     "lerna": "^3.20.2",
     "lint-staged": "^10.0.8",
-    "next": "^9.3.0",
+    "next": "^9.3.7",
```
Comment (merelinguist) [inline package.json:76]: Thanks for the PR! Any reason not to just set this to the canary version? `9.3.7-canary.10`

Comment (dwightwatson) [reply package.json:76]: Can do that - just thought it would be weird to commit a canary version. Would it be better to just do that and get on with it?

Comment (flybayer) [reply package.json:76]: I'm fine to do this. Also, we need to pin the next.js version. It shouldn't be a range.

**Comment by dwightwatson** [comment]
And we're back to green 🎉

**Comment by flybayer** [comment]
Great, thank you!!

@all-contributors add @dwightwatson for code

**Pull request merged**